tools/hasher/src/main/java/org/apache/shiro/tools/hasher/Hasher.java

```diff
@@ -504,6 +507,15 @@ private static char[] readPassword(boolean confirm) throws IOException {
         return first;
     }
 
+    private static boolean isTerminal(java.io.Console console) throws IOException {
+        try {
+            // isTerminal is only available in Java 22 or later
+            return console != null && (Boolean) PropertyUtils.getProperty(console, "terminal");
```

Comment:
Can we use standard reflection here? I dont want to introduce a dependency just for one line.

Also, newer JDKs will bring better Reflection tools iirc.

Reply:
done